Assets/Scripts/Board/Graph.cs

```diff
@@ -0,0 +1,80 @@
+using UnityEngine;
+using System.Collections.Generic;
+
+public class Graph
+{
+    public enum Building
+    {
+        Settlement,
+        City,
+    }
+
+    public class Node { }
```

Comment:
The main advantage of having a Node base class is to either (1) place common logic between all nodes, or (2) make it possible to have a List<Node> that contains both VertexNode/EdgeNode objects. Maybe we don't have a reason to do these things yet and will in the future, but worth noting. 

(1) could include
* `Building` if `Building` included Road and Robber
* the Vector3Int and/or Vector3 which store the node's position, that way it can be accessed and used by the Board class to get the world position in the tilemap 

Just food for thought, nothing actionable per se and not sure if it's even a good thought.

Reply:
I added a `worldPosition` field to `Node`, which feels helpful. As far as augmenting `Building` with road and robber, I'm not sure I like that, since the type of building that a node can have would then be limited by the node type, but that isn't represented by having all them in the `Building` enum.